From 977146f647ff47086003a1c24890d516a52e2983 Mon Sep 17 00:00:00 2001
From: Elvis Lee <kwangwoong.lee@lge.com>
Date: Wed, 4 Nov 2020 17:34:05 +0900
Subject: [PATCH] Support customized device discovery with udev

With derived class from QDeviceDiscoveryUDev, eglfs for a platform
can create own input handlers.

Task-number: QTBUG-85268
Change-Id: Ib8e99b365195eff9540e7c90e14c5d269d37ad86
Reviewed-by: Elvis Lee <kwangwoong.lee@lge.com>
Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
Upstream-Status: Backport [https://codereview.qt-project.org/c/qt/qtbase/+/320279]
---
 .../devicediscovery/qdevicediscovery_udev_p.h               | 4 +++-
 src/plugins/platforms/eglfs/api/qeglfsintegration_p.h       | 6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/platformsupport/devicediscovery/qdevicediscovery_udev_p.h b/src/platformsupport/devicediscovery/qdevicediscovery_udev_p.h
index 28618d0b21..0ab5917738 100644
--- a/src/platformsupport/devicediscovery/qdevicediscovery_udev_p.h
+++ b/src/platformsupport/devicediscovery/qdevicediscovery_udev_p.h
@@ -68,13 +68,15 @@ public:
 private slots:
     void handleUDevNotification();
 
+protected:
+    struct udev *m_udev;
+
 private:
     bool checkDeviceType(struct udev_device *dev);
 
     void startWatching();
     void stopWatching();
 
-    struct udev *m_udev;
     struct udev_monitor *m_udevMonitor;
     int m_udevMonitorFileDescriptor;
     QSocketNotifier *m_udevSocketNotifier;
diff --git a/src/plugins/platforms/eglfs/api/qeglfsintegration_p.h b/src/plugins/platforms/eglfs/api/qeglfsintegration_p.h
index 898b322834..d2810eb1ed 100644
--- a/src/plugins/platforms/eglfs/api/qeglfsintegration_p.h
+++ b/src/plugins/platforms/eglfs/api/qeglfsintegration_p.h
@@ -103,9 +103,12 @@ public:
 
     QFbVtHandler *vtHandler() { return m_vtHandler.data(); }
 
+protected:
+    virtual void createInputHandlers();
+    QEvdevKeyboardManager *m_kbdMgr;
+
 private:
     EGLNativeDisplayType nativeDisplay() const;
-    void createInputHandlers();
     static void loadKeymapStatic(const QString &filename);
     static void switchLangStatic();
 
@@ -114,7 +117,6 @@ private:
     QScopedPointer<QPlatformFontDatabase> m_fontDb;
     QScopedPointer<QPlatformServices> m_services;
     QScopedPointer<QFbVtHandler> m_vtHandler;
-    QEvdevKeyboardManager *m_kbdMgr;
     bool m_disableInputHandlers;
 };
 
